lapce-ui/src/terminal.rs

```diff
@@ -519,9 +551,55 @@ impl Widget<LapceTabData> for LapceTerminal {
             config: data.config.clone(),
             find: data.find.clone(),
         };
+        ctx.set_cursor(&Cursor::IBeam);
         match event {
-            Event::MouseDown(_mouse_event) => {
+            Event::MouseDown(mouse_event) => {
+                let now = Instant::now();
                 self.request_focus(ctx, data);
+                let terminal = data.terminal.terminals.get(&self.term_id).unwrap();
+                let term = &mut terminal.raw.lock().term;
+                if mouse_event.button.is_right() {
+                    let mut clipboard = SystemClipboard {};
+                    match term.selection_to_string() {
+                        Some(selection) => {
+                            clipboard.put_string(selection);
+                            term.selection = None;
+                        }
+                        None => match clipboard.get_string() {
+                            Some(string) => {
+                                terminal.proxy.proxy_rpc.terminal_write(
+                                    terminal.term_id,
+                                    string.as_str(),
+                                );
+                                term.scroll_display(Scroll::Bottom);
+                            }
+                            None => {}
+                        },
+                    }
+                } else {
+                    term.selection = None;
+                    if self.prev_click != None
+                        && (now - self.prev_click.unwrap()).as_millis()
+                            < DOUBLE_CLICK_TIMEOUT
```

Comment:
You can use ```mouse_event.count``` to know if it's a double click, which should be easier.

Reply:
Good catch! I should be able to implement triple-click line select then!